### PR TITLE
Fixes #14899: improved output of "Eval red in term" by betaiota-normalizing the type, like Check does

### LINKED ISSUE #14899: Missing reduction in type printed by `Compute` command
#### Description of the problem

The `Compute` command does less reduction on the type of its input than the `Check` command.  This makes the output of `Compute` harder to read than necessary:

```coq
Definition a : { x | 0 < x }.
  exists 3. eauto.
Defined.

Check (proj2_sig a).
(* proj2_sig a
     : 0 < proj1_sig a *)

Compute (proj2_sig a).
(* = le_S 1 2 (le_S 1 1 (le_n 1))
     : (fun x : nat => 0 < x) (proj1_sig a) *)
```

Here is another example:

```
Definition hd {A} (l: list A) :  match l with [] => unit: Type | _ => A end :=
  match l return (match l with [] => unit | _ => A end) with
  | [] => tt
  | h :: tl => h
  end.

Compute hd [1].
(* = 1
   : match [1] with
     | [] => unit : Type
     | _ :: _ => Z
     end *)
```

It would be nice if Compute reduced the type of the output, too.  I'm not sure whether that would break anything.

#### Coq Version

```console
$ coqtop -v
The Coq Proof Assistant, version 8.13.2 (June 2021)
compiled on Jun 1 2021 3:28:22 with OCaml 4.09.0
```

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** enhancement/fix

The betaiota-normalization was done for `Check`/`Type` but not for `Eval-in`/`Compute`.

Closes #14899
                                                                                                                                                                                                                                                                                        
 - [x] Added **changelog**.

### CODE REVIEW

File: vernac/vernacentries.ml [hunk 1]
```diff
@@ -1849,11 +1849,11 @@ let vernac_check_may_eval ~pstate redexp glopt rc =
       (* OK to call kernel which does not support evars *)
       Environ.on_judgment EConstr.of_constr (Arguments_renaming.rename_typing env c)
   in
+  let j = { j with Environ.uj_type = Reductionops.nf_betaiota env sigma j.Environ.uj_type } in
   let pp = match redexp with
     | None ->
         let evars_of_term c = Evarutil.undefined_evars_of_term sigma c in
         let l = Evar.Set.union (evars_of_term j.Environ.uj_val) (evars_of_term j.Environ.uj_type) in
-        let j = { j with Environ.uj_type = Reductionops.nf_betaiota env sigma j.Environ.uj_type } in
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml:1856]: I guess it makes more sense to use the evars_of_term of the type which actually gets printed too.

Comment (herbelin) [reply vernac/vernacentries.ml:1856]: Do you mean that it is incidentally a (theoretical) improvement in the `Check` case too, where evars of the type which may have disappeared by reduction are now not printed?

Comment (SkySkimmer) [reply vernac/vernacentries.ml:1856]: yes

File: doc/changelog/07-vernac-commands-and-options/14901-master+wish14899-nf-betaiota-on-result-evai-in.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+- **Fixed:**
+  :cmd:`Eval` and :cmd:`Compute term` now beta-iota-simplify the type
+       of the result, like :cmd:`Check` and :cmd:`Type` do
```
Comment (SkySkimmer) [inline doc/changelog/07-vernac-commands-and-options/14901-master+wish14899-nf-betaiota-on-result-evai-in.rst:3]: bad indent

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@herbelin the changelog is broken, you need to fix it.

**Comment by ppedrot** [comment]
@herbelin still broken!

**Pull request merged**